src/main/java/org/folio/rest/impl/OaiPmhViewInstancesAPI.java

```diff
@@ -36,23 +42,33 @@
     .lookupClass());
 
   private static final String SQL = "select * from pmh_view_function($1,$2,$3,$4);";
+  private static final String SQL_UPDATED_INSTANCES_IDS = "select * from pmh_get_updated_instances_ids($1,$2,$3,$4);";
+  private static final String SQL_INSTANCES = "select * from pmh_instance_view_function($1,$2);";
 
+  @Validate
   @Override
   public void getOaiPmhViewInstances(String startDate, String endDate, boolean deletedRecordSupport,
       boolean skipSuppressedFromDiscoveryRecords, String lang, RoutingContext routingContext, Map<String, String> okapiHeaders,
       Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
 
-    log.debug("request params:", Iterables.toString(routingContext.request()
-      .params()));
+    getCommonInstanceProcessing(SQL,
+      () -> createPostgresParams(startDate, endDate, deletedRecordSupport, skipSuppressedFromDiscoveryRecords),
+      routingContext, okapiHeaders, asyncResultHandler, vertxContext,
+      "Select from oai pmh view completed successfully");
+  }
+
+  private void getCommonInstanceProcessing(String sql, Supplier<Tuple> paramsSupplier, RoutingContext routingContext, Map<String, String> okapiHeaders,
+      Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext, String logMessage) {
 
     try {
-      Tuple params = createPostgresParams(startDate, endDate, deletedRecordSupport, skipSuppressedFromDiscoveryRecords);
+      log.debug("request params:", Iterables.toString(routingContext.request().params()));
```

Comment:
```suggestion
      if (log.isDebugEnabled()) {
         log.debug("request params:", Iterables.toString(routingContext.request().params()));
      }
```

Reply:
Possibly it would be event better to avoud double ckecking of debug enabling and just use parametrized messages, which in turns uses lazy message formatting, I mean after detecting that debug leven is enabed.